common/big_messaging/src/main/scala/uk/ac/wellcome/bigmessaging/BigMessageReader.scala

```diff
@@ -2,35 +2,37 @@ package uk.ac.wellcome.bigmessaging
 
 import grizzled.slf4j.Logging
 import io.circe.Decoder
-import uk.ac.wellcome.bigmessaging.message.{
-  InlineNotification,
-  MessageNotification,
-  RemoteNotification
-}
 import uk.ac.wellcome.json.JsonUtil.fromJson
 import uk.ac.wellcome.bigmessaging.message.{
   InlineNotification,
+  MessageNotification,
   RemoteNotification
 }
-import uk.ac.wellcome.storage.ObjectStore
+import uk.ac.wellcome.storage.{Identified, NotFoundError, ObjectLocation}
+import uk.ac.wellcome.storage.store.{TypedStore, TypedStoreEntry}
 
 import scala.util.{Failure, Success, Try}
 
 trait BigMessageReader[T] extends Logging {
-  val objectStore: ObjectStore[T]
+  val typedStore: TypedStore[ObjectLocation, T]
 
   implicit val decoder: Decoder[T]
 
+  // TODO: Turn this into an Either[ReadError, T]
```

Comment:
Maybe some context in the comment of why it would be desirable for this to return  `Either[ReadError, T]`?

Reply:
```suggestion
  // TODO: Turn this into an Either[ReadError, T] to match the underlying pattern in TypedStore
  // See: https://github.com/wellcometrust/scala-storage/blob/master/storage/src/main/scala/uk/ac/wellcome/storage/store/Store.scala#L8
```